### PR TITLE
Agent.build now honors embedded_path for rtloader

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Currently building the Agent with `agent.build` will not work properly with custom `embedded_path` as implicitly built rtloader will not be installed there but in `./dev` anyway.
It's useful to build Agent binaries that can be hotswapped easily in containers.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

No QA

### CODE REVIEW

**Review by sgnn7 — COMMENTED** [review]
@vboulineau Do you mind adding some info on this to https://github.com/DataDog/datadog-agent/blob/master/docs/dev/agent_dev_env.md?

**Comment by vboulineau** [comment]
> @vboulineau Do you mind adding some info on this to https://github.com/DataDog/datadog-agent/blob/master/docs/dev/agent_dev_env.md?

Added a doc about target usage of this change

File: docs/dev/agent_build.md [hunk 1]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
 to adjust the build for your specific configuration, add or edit the files within
 the `pkg-config` folder.
 
+## Testing Agent changes in containerized environments
+
+Building an Agent Docker image from strach through an embedded build is a slow process.
```
Comment (mx-psi) [inline docs/dev/agent_build.md:56]:
```suggestion
Building an Agent Docker image from scratch through an embedded build is a slow process.
```

File: docs/dev/agent_build.md [hunk 2]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
...
+Sometimes, you may want to quickly test a change or bufix in a containerized environment (Docker, Kubernetes, ECS, etc.).
```
Comment (mx-psi) [inline docs/dev/agent_build.md:57]:
```suggestion
Sometimes, you may want to quickly test a change or bugfix in a containerized environment (Docker, Kubernetes, ECS, etc.).
```

**Review by mx-psi — DISMISSED** [review]

**Comment by mx-psi** [comment]
woops, meant to be a comment, not an approval

**Review by mx-psi — APPROVED** [review]
LGTM, CI errors are probably fixed by merging/rebasing master

File: docs/dev/agent_build.md [hunk 3]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
 to adjust the build for your specific configuration, add or edit the files within
 the `pkg-config` folder.
 
+## Testing Agent changes in containerized environments
+
+Building an Agent Docker image from scratch through an embedded build is a slow process.
+Sometimes, you may want to quickly test a change or bugfix in a containerized environment (Docker, Kubernetes, ECS, etc.).
```
Comment (jtappa) [inline docs/dev/agent_build.md:57]:
```suggestion
You can quickly test a change or bug fix in a containerized environment (such as Docker, Kubernetes, or ECS).
```

File: docs/dev/agent_build.md [hunk 4]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
...
+
+One way to do it is to patch the Agent binary from official Docker image, with a simple Dockerfile:
```
Comment (jtappa) [inline docs/dev/agent_build.md:59]:
```suggestion
One way to do this is to patch the Agent binary from an official Docker image, with a Dockerfile:
```

File: docs/dev/agent_build.md [hunk 5]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
...
+
+```
+FROM datadog/agent:<AGENT_VERSION>
+
+COPY agent /opt/datadog-agent/bin/agent/agent
+```
+
+However, for this to work properly, two things are important:
+- Your change needs to be done on top of `<AGENT_VERSION>` tag from our repository.
```
Comment (jtappa) [inline docs/dev/agent_build.md:68]:
```suggestion
- Your change needs to be done on top of the `<AGENT_VERSION>` tag from the DataDog repository.
```

File: docs/dev/agent_build.md [hunk 6]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
...
+- You need to run invoke task with proper embedded path `inv -e agent.build -e /opt/datadog-agent/embedded`.
```
Comment (jtappa) [inline docs/dev/agent_build.md:69]:
```suggestion
- You need to run the invoke task with the proper embedded path `inv -e agent.build -e /opt/datadog-agent/embedded`.
```

File: docs/dev/agent_build.md [hunk 7]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
...
+
+Note that this will make `invoke` to install the builds artifacts in `/opt/datadog-agent/embedded` folder. Make sure folder exists and current user has write permissions.
```
Comment (jtappa) [inline docs/dev/agent_build.md:71]:
```suggestion
**Note**: This makes `invoke` install the build's artifacts in the `/opt/datadog-agent/embedded` folder. Make sure the folder exists and the current user has write permissions.
```

File: docs/dev/agent_build.md [hunk 8]
```diff
@@ -51,4 +51,23 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
...
+
+One way to do it is to patch the Agent binary from official Docker image, with a simple Dockerfile:
+
+```
+FROM datadog/agent:<AGENT_VERSION>
+
+COPY agent /opt/datadog-agent/bin/agent/agent
+```
+
+However, for this to work properly, two things are important:
```
Comment (jtappa) [inline docs/dev/agent_build.md:67]:
```suggestion
For this to work properly, two things are important:
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Review by jtappa — APPROVED** [review]

**Pull request merged**